src/main/scala/rocket/ScratchpadSlavePort.scala

```diff
@@ -97,22 +96,25 @@ class ScratchpadSlavePort(address: AddressSet)(implicit p: Parameters) extends L
 }
 
 /** Mix-ins for constructing tiles that have optional scratchpads */
-trait CanHaveScratchpad extends HasHellaCache with HasICacheFrontend with HasCoreParameters {
+trait CanHaveScratchpad extends HasHellaCache with HasICacheFrontend {
   val module: CanHaveScratchpadModule
+  val xLenBytes = p(XLen)/8
```

Comment:
Since XLen means "length of an x-register" maybe this could be ```xBytes```?  I figure it will show up a lot, so might as well pick a good name now.